internal/filters/filters.go

```diff
@@ -0,0 +1,134 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package filters
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/utils/strings/slices"
+)
+
+var (
+	// ValidTypes are the valid types of Elastic resources that are supported by the filtering system.
+	ValidTypes              = []string{"agent", "apm", "beat", "elasticsearch", "enterprisesearch", "kibana", "maps"}
+	elasticTypeKey          = "common.k8s.elastic.co/type"
+	elasticsearchNameFormat = "%s.k8s.elastic.co/cluster-name"
+	elasticNameFormat       = "%s.k8s.elastic.co/name"
+)
+
+// Filter is a type + name filter that translates into a labelSelector that is applied when querying for Kubernetes resources.
+// Both type and name are required at this time.
+//
+// examples supported:
+// name=mycluster, type=elasticsearch
+// name=mykb, type=kibana
+type Filter struct {
+	source        []string
+	typ           string
+	name          string
+	labelSelector string
+}
+
+// LabelSelector returns the formatted labelSelector for the filter.
+func (f Filter) LabelSelector() string {
+	return f.labelSelector
+}
+
+// Type returns the type of the Elastic resources to filter.
+func (f Filter) Type() string {
+	return f.typ
+}
+
+// Name returns the name of the Elastic resource to filter.
+func (f Filter) Name() string {
+	return f.name
+}
+
+// New returns a new filter, given a slice of key=value pairs,
+// runs validation on the given slice, and returns an error
+// if the given key=value pairs are invalid.
+//
+// source example:
+// []string{"name=mycluster", "type=elasticsearch"}
+func New(source []string) (Filter, error) {
+	filter := Filter{
+		source: source,
+	}
+	return filter.validate()
+}
```

Comment:
```suggestion
func New(source []string) (Filter, error) {	
	return parse(source)
}
```
or inline parse into `New`